pkg/commands/loaders/branches_test.go

```diff
@@ -0,0 +1,72 @@
+package loaders
+
+// "*|feat/detect-purge|origin/feat/detect-purge|[ahead 1]"
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestObtainBranchTrimHeads(t *testing.T) {
```

Comment:
For the sake of keeping these tests DRY, could we structure this file like so:
```go
func TestObtainBranch(t *testing.T) {
	type scenario struct {
		testName string
		input []string
		expectedBranch *models.Branch
	}

	scenarios := []scenario{
		...
	}

	for _, s := range scenarios {
		t.Run(s.testName, func(t *testing.T) {
			branch := obtainBranch(split)
			assert.EqualValues(t, s.expectedBranch, branch.Name)
		})
	}
}
```

Reply:
Sure. I can tackle the test refactor. I will open another PR to your feature branch @john-nguyen09.

Reply:
@john-nguyen09 I have opened the relevant [PR #2](https://github.com/john-nguyen09/lazygit/pull/2) to your feature branch

Reply:
Thanks @abid-mujtaba, the PR is merged.